source/shared/cpp/ObjectModel/MarkDownParser.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <iomanip>
+#include <list>
+#include "pch.h"
+#include "BaseCardElement.h"
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+class MarkDownParser
+{
+public:
+    MarkDownParser (const std::string &txt) : m_currentDelimiterType (Init), 
+        m_currentWordIndex(0), m_lookBehind(Init), m_delimiterCnts(0), m_EmphasisState (EmphasisNotDetected), 
+        m_checkLookAhead(false), m_checkIntraWord(false), m_isRightEmphasisDetecting(false), m_text(txt), 
+        m_textBegin(m_text.begin()), m_curPos(m_text.begin()), m_textEnd(m_text.end()), m_LeftEmphasisDetecting(0),
+        m_leftLookUpTable (), m_rightLookUpTable (), m_tokenizedString(1, "") {};
```

Comment:
Sorry if there was some confusion in my previous comment.

Any members that are always initialized with the same values, should be initialized in the header during declaration:
m_checkLookAhead = false

Keep the members that depend on the passed in string, like m_text and m_textBegin in the cpp file.